src/main/java/com/car/app/controller/CarDao.java

```diff
@@ -100,4 +120,16 @@ public boolean deleteCar(String carId) {
 			return false;
 		}
 	}
+	
+	static public String filteringAndSortingQuery(String sort, String order, String search, String searchBy) {
+		
+		String query = "SELECT c  FROM Car c ";
```

Comment:
Dealing with strings in dynamic queries is a dangerous approach, please check Criteria API https://www.objectdb.com/java/jpa/query/criteria .